### PR TITLE
Added two new methods to fetch Party and List<Party> based on uuid

### PR DESCRIPTION
**Pull request by lovoll**
also fixed the GetParty to fetch PartyUuid by setting option for case insensitive property name as it is defined as PartyUUID in SblBridge and PartyUuid in Altinn.Platform.Models

## Related Issue(s)
- #{issue number}

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by SandGrainOne — CHANGES_REQUESTED** [review]

File: src/Controllers/PartiesController.cs [hunk 1]
```diff
@@ -125,6 +148,28 @@ public async Task<ActionResult<List<Party>>> GetPartyListForPartyIds([FromBody]
...
+        [Produces("application/json")]
+        public async Task<ActionResult<List<Party>>> GetPartyListForPartyUuids([FromBody] List<Guid> partyUuids)
+        {
+            List<Party> parties;
+
+            parties = await _partiesWrapper.GetPartyListByUuid(partyUuids);
+
+            if (parties == null)
+            {
+                return NotFound();
```
[the remark this answers is not in the document]
Comment (lovoll) [reply src/Controllers/PartiesController.cs]: Fixed

File: src/Controllers/PartiesController.cs [hunk 2]
```diff
@@ -125,6 +148,28 @@ public async Task<ActionResult<List<Party>>> GetPartyListForPartyIds([FromBody]
...
+        /// <param name="partyUuids">List of partyUuids for parties to retrieve.</param>
+        /// <returns>List of parties based on the party uuids.</returns>
+        [HttpPost("partylistbyuuid")]
+        [Consumes("application/json")]
+        [Produces("application/json")]
+        public async Task<ActionResult<List<Party>>> GetPartyListForPartyUuids([FromBody] List<Guid> partyUuids)
+        {
+            List<Party> parties;
+
+            parties = await _partiesWrapper.GetPartyListByUuid(partyUuids);
```
[the remark this answers is not in the document]
Comment (lovoll) [reply src/Controllers/PartiesController.cs]: Fixed

**Review by SandGrainOne — APPROVED** [review]

File: src/Controllers/PartiesController.cs [hunk 3]
```diff
@@ -77,6 +77,29 @@ public async Task<ActionResult<Party>> Get(int partyId)
...
+        /// <summary>
+        /// Gets the party for a given party uuid.
+        /// </summary>
+        /// <param name="partyUuid">The party uuid.</param>
+        /// <returns>The information about a given party.</returns>
+        [HttpGet("{partyUuid:Guid}")]
+        [ProducesResponseType(404)]
+        [ProducesResponseType(200)]
+        [Produces("application/json")]
+        public async Task<ActionResult<Party>> GetByUuid(Guid partyUuid)
```
Comment (SandGrainOne) [inline src/Controllers/PartiesController.cs:89]: Consider adding `[Authorize]` and the same validation as in the `Get(int partyId)` method above.

File: test/UnitTests/PartyWrapperTest.cs [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+            Assert.True(inCache);
+            Assert.Equal(partyUuid, fromCache.PartyUuid);
+
+            Assert.NotNull(sblRequest);
+            Assert.Equal(HttpMethod.Get, sblRequest.Method);
+            Assert.EndsWith($"parties?partyuuid=4c3b4909-eb17-45d5-bde1-256e065e196a", sblRequest.RequestUri!.ToString());
+        }
+
+        [Fact]
+        public async Task GetParty_SblBridge_finds_party_Target_returns_NotFound()
```
[the remark this answers is not in the document]
Comment (lovoll) [reply test/UnitTests/PartyWrapperTest.cs]: Renamed

File: test/UnitTests/PartyWrapperTest.cs [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+            Assert.Equal(partyUuids[1], actual[1].PartyUuid);
+            Assert.Equal(partyUuids[2], actual[2].PartyUuid);
+
+            Assert.NotNull(sblRequest);
+            Assert.Equal(HttpMethod.Post, sblRequest.Method);
+            Assert.EndsWith($"parties/byuuid", sblRequest.RequestUri!.ToString());
+        }
+
+        [Fact]
+        public async Task GetParty_SblBridge_finds_partylist_Target_returns_NotFound()
```
[the remark this answers is not in the document]
Comment (lovoll) [reply test/UnitTests/PartyWrapperTest.cs]: Renamed

**Comment by SandGrainOne** [comment]
Looks good!

**Pull request merged**